openstack/csms/v1/secrets/requests.go

```diff
@@ -0,0 +1,133 @@
+package secrets
+
+import "github.com/chnsz/golangsdk"
+
+var RequestOpts = golangsdk.RequestOpts{
+	MoreHeaders: map[string]string{"Content-Type": "application/json", "X-Language": "en-us"},
+}
+
+type CreateSecretOpts struct {
+	Name         string `json:"name" required:"true"`
+	KmsKeyID     string `json:"kms_key_id,omitempty"`
+	Description  string `json:"description,omitempty"`
+	SecretBinary string `json:"secret_binary,omitempty"`
+	SecretString string `json:"secret_string,omitempty"`
+}
+
+func Create(c *golangsdk.ServiceClient, opts CreateSecretOpts) (*Secret, error) {
+	b, err := golangsdk.BuildRequestBody(opts, "")
+	if err != nil {
+		return nil, err
+	}
+
+	var rst golangsdk.Result
+	_, err = c.Post(rootURL(c), b, &rst.Body, &golangsdk.RequestOpts{
+		MoreHeaders: RequestOpts.MoreHeaders,
+	})
+	if err == nil {
+		var r SecretRst
+		rst.ExtractInto(&r)
+		return &r.Secret, nil
+	}
+	return nil, err
+}
+
+func Get(c *golangsdk.ServiceClient, secretName string) (*Secret, error) {
+	var rst golangsdk.Result
+	_, err := c.Get(resourceURL(c, secretName), &rst.Body, &golangsdk.RequestOpts{
+		MoreHeaders: RequestOpts.MoreHeaders,
+	})
+	if err == nil {
+		var r SecretRst
+		rst.ExtractInto(&r)
+		return &r.Secret, nil
+	}
+	return nil, err
+}
+
+type UpdateSecretOpts struct {
+	KmsKeyID    string  `json:"kms_key_id,omitempty"`
```

Comment:
can `kms_key_id` be updated empty which means using the default kms?

Reply:
If the value is empy, the API returned an error.
If we want to change it, the value must be specified,  even if we change to the default kms.